pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -80,6 +89,45 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 	return nil
 }
 
+func (l *listenerSynthesizer) getLatticeListenerDefaultAction(ctx context.Context, stackListener *model.Listener) (
```

Comment:
The conversion logic from model -> Lattice type typically lives in each component's `manager`. Here, it should only resolve the target group Ids.

Please move `getDefaultAction` logic inside `listener_manager`. The logic should assume the target group Ids have already been resolved. However, it needs to handling the case where the target group ID is set to `model.InvalidBackendRefTgId`. This can happen when the `backendRef` points to something that doesn't exist. Note we should have a test specifically for `InvalidBackendRefTgId` handling, which I don't think we do yet since I don't see it referenced here.

Reply:
Working on in. Will do the exactly you describe here in the new revision. Will mimic the `r *defaultRuleManager) buildLatticeRule()` to get the listener defaultAction

 https://github.com/aws/aws-application-networking-k8s/blob/3e6585687487fe3339063b610527403b2dab183a/pkg/deploy/lattice/rule_manager.go#L92-L143

Reply:
Sent the new revision can you take a look? Did the changes exactly as you describe above.  Thank you! https://github.com/aws/aws-application-networking-k8s/pull/643/commits/b4dfd4dedcbe6cfce9d935f3477dcf5e2e29b203

> Note we should have a test specifically for InvalidBackendRefTgId handling
The test cases: `Test_ListenerManager_getLatticeListenerDefaultAction_TLS_PASSTHROUGH_Listener` covered that.


Also ran the full e2e test suite against new revision
```
[SynchronizedAfterSuite]
/Volumes/workspace/aws-application-networking-k8s/test/suites/integration/suite_test.go:72
{“level”:“info”,“ts”:“2024-06-07T01:26:52.316-0700",“caller”:“test/framework.go:264",“msg”:“Deleting objects: *v1.Gateway/test-gateway, *v1.Pod/grpc-runner”}
{“level”:“info”,“ts”:“2024-06-07T01:26:52.357-0700",“caller”:“test/framework.go:282",“msg”:“Waiting for NotFound, objects: *v1.Gateway/test-gateway, *v1.Pod/grpc-runner”}
[SynchronizedAfterSuite] PASSED [31.041 seconds]
------------------------------
Ran 66 of 69 Specs in 2407.814 seconds
SUCCESS! -- 66 Passed | 0 Failed | 0 Pending | 3 Skipped
--- PASS: TestIntegration (2407.82s)
PASS
ok 	[github.com/aws/aws-application-networking-k8s/test/suites/integration](http://github.com/aws/aws-application-networking-k8s/test/suites/integration)	2408.753s
```